src/legacy/core_plugins/kibana/public/dashboard/np_ready/dashboard_app_controller.tsx

```diff
@@ -170,9 +164,20 @@ export class DashboardAppController {
 
     // The hash check is so we only update the time filter on dashboard open, not during
     // normal cross app navigation.
-    if (dashboardStateManager.getIsTimeSavedWithDashboard() && !hasInheritedGlobalStateFromUrl) {
+    const initialGlobalStateInUrl = kbnUrlStateStorage.get<QueryState>('_g');
+    const hasInheritedTimeRange = Boolean(initialGlobalStateInUrl?.time);
```

Comment:
This fixes restoring of time range in saved dashboard after drilldown. 
The bug was that `hasInheritedGlobalStateFromUrl` looked only on present `_g` in the url. but `time` could be missing there. So now we specifically look for `_g.time` to decide if we need to use dashboard's time as initial time

Reply:
Extracted to separate pr in scope of: https://github.com/elastic/kibana/issues/61230